File: src/javaapp/AdminUi.java
package javaapp;

import java.util.Scanner;

public class AdminUi implements UserInterface{

    private Admin admin;

    public AdminUi(Admin admin) {
        this.admin = admin;
    }

    @Override
    public void displayMenu() {
        System.out.println("\nAdmin Menu:");
        System.out.println("1. Add New Product");
        System.out.println("2. Update Product");
        System.out.println("3. Remove Product");
        System.out.println("4. View All Products");
        System.out.println("0. Back to Main Menu");
    }

    @Override
    public void handleUserInput() {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter your choice: ");
        int choice = scanner.nextInt();
        switch (choice) {
            case 1:
                admin.addNewProduct();
                break;
            case 2 :
                admin.updateProduct();
                break;
            case 3 :
                System.out.print("Enter Product Id : ");
                String productId = scanner.nextLine();
                admin.removeProduct(productId);
                break;
            case 4:
                admin.viewAllProducts();
                break;
            case 0:
                return;
            default:
                System.out.print("Invalid input. Please enter a number.");
                break;
        }
    }
}

File: src/kotlinapp/AdminUi.kt
package kotlinapp

import javaapp.Admin
import javaapp.UserInterface
import java.util.*

class AdminUi(private val admin: Admin) : UserInterface {
    override fun displayMenu() {
        println("\nAdmin Menu:")
        println("1. Add New Product")
        println("2. Update Product")
        println("3. Remove Product")
        println("4. View All Products")
        println("0. Back to Main Menu")
    }

    override fun handleUserInput() {
        val scanner = Scanner(System.`in`)
        print("Enter your choice: ")
        val choice = scanner.nextInt()
        when (choice) {
            1 -> admin.addNewProduct()
            2 -> admin.updateProduct()
            3 -> {
                print("Enter Product Id : ")
                val productId = scanner.nextLine()
                admin.removeProduct(productId)
            }
            4 -> admin.viewAllProducts()
            0 -> return
            else -> print("Invalid input. Please enter a number.")
        }
    }
}


